Lib/test/test_frame.py

```diff
@@ -397,15 +397,41 @@ def test_repr(self):
     def test_delete(self):
         x = 1
         d = sys._getframe().f_locals
-        with self.assertRaises(TypeError):
+
+        # This needs to be tested before f_extra_locals is created
+        with self.assertRaises(KeyError):
+            del d['non_exist']
+
+        with self.assertRaises(KeyError):
+            d.pop('non_exist')
```

Comment:
Consider adding a test with `assertRaisesRegex` to make sure that the contents of the `KeyError` is correct.